### PR TITLE
Switch Logging to Slf4j

### PR DESCRIPTION
**Pull request by bulldog98**
This changes the logging to slf4j as discussed in #251, it also implement the checkstyle check, as suggested in https://github.com/refinedmods/refinedstorage2/issues/251#issuecomment-1368095194

### CODE REVIEW

**Comment by bulldog98** [comment]
I do not think this one needs a changelog, but I'm unable to add the skip-changelog to it

File: refinedstorage2-grid-api/build.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 archivesBaseName = 'refinedstorage2-grid-api'
 
 dependencies {
-    implementation libs.log4j.api
+    implementation libs.slf4j.api
```
Comment (raoulvdberge) [inline refinedstorage2-grid-api/build.gradle:4]: If we have a dependency on slf4j here, we probably need a testRuntimeOnly slf4j-simple here as well to get logging.

File: refinedstorage2-network-api/build.gradle [hunk 2]
```diff
@@ -1,8 +1,8 @@
 archivesBaseName = 'refinedstorage2-network-api'
 
 dependencies {
-    implementation libs.log4j.api
-    testImplementation libs.log4j.core
+    implementation libs.slf4j.api
+    testImplementation libs.slf4j.impl
```
Comment (raoulvdberge) [inline refinedstorage2-network-api/build.gradle]: This should probably be testRuntimeOnly

File: settings.gradle [hunk 3]
```diff
@@ -92,6 +92,10 @@ dependencyResolutionManagement {
             def log4j = version("log4j", "2.17.0")
             library("log4j-api", "org.apache.logging.log4j", "log4j-api").versionRef(log4j)
             library("log4j-core", "org.apache.logging.log4j", "log4j-core").versionRef(log4j)
```
Comment (raoulvdberge) [inline settings.gradle]: Can we remove log4j from here?

File: settings.gradle [hunk 4]
```diff
@@ -92,6 +92,10 @@ dependencyResolutionManagement {
...
+            def slf4j = version("slf4j", "1.8.0-beta4")
+            library("slf4j-api", "org.slf4j", "slf4j-api").versionRef(slf4j)
+            library("slf4j-impl", "org.slf4j", "slf4j-simple").versionRef(slf4j)
+            bundle("slf4j-impl", ["slf4j-api", "log4j-api", "log4j-core", "slf4j-impl"])
```
Comment (raoulvdberge) [inline settings.gradle]: I don't think a bundle really works here as we want runtimeOnly specifically for slf4j-simple?

Comment (raoulvdberge) [inline settings.gradle]: I think log4j can be removed here as well?

**Review by raoulvdberge — CHANGES_REQUESTED** [review]

File: settings.gradle (hunk 3 above)
Comment (bulldog98) [reply settings.gradle]: sure, I just forgot it.

File: settings.gradle (hunk 4 above)
Comment (bulldog98) [reply settings.gradle]: yes, that is also unused, as far as I remember

File: refinedstorage2-grid-api/build.gradle (hunk 1 above)
Comment (bulldog98) [reply refinedstorage2-grid-api/build.gradle:4]: Added that

**Review by raoulvdberge — APPROVED** [review]

**Comment by raoulvdberge** [comment]
Thanks, good stuff!

**Pull request merged**